src/lib/wallet/IndyWallet.ts

```diff
@@ -30,34 +31,28 @@ export class IndyWallet implements Wallet {
     logger.log(`Wallet opened with handle: ${this.wh}`);
   }
 
-  async initPublicDid(did: Did, seed: string) {
-    let didInfo = {};
-    try {
-      const existingVerkey = await this.keyForLocalDid(did);
-      // TODO validate that verkey is as same as the verkey generated by given seed
-      didInfo = {
-        did,
-        verkey: existingVerkey,
-      };
-    } catch (error) {
-      if (error.name === 'IndyError' && error.message === '212') {
-        // WalletItemNotFound
-        logger.log('WalletItemNotFound');
-        const [_, verkey] = await this.createDid({ did, seed });
-        didInfo = {
-          did,
-          verkey,
-        };
-      } else {
-        throw error;
-      }
-    }
+  async initAgentDid(did: Did, seed: string) {
```

Comment:
A DID is generated from a seed right? so passing DID shouldn't be necessary

Reply:
Yes, that's a good point, but I think it could cause changes outside the scope of this PR so I would rather do it in a different one.

Reply:
I got rid of `Wallet`'s `initAgentDid` and `Agent`'s `initPublicDid` method. Now, the `initPublicDid` is called inside `Agent.init()` as it was before, I just refactored method signature to be more benevolent and take `DidInfo`.